dapr_agents/observability/instrumentor.py

```diff
@@ -0,0 +1,516 @@
+            logger.debug(
+                "🔧 Monkey patching WorkflowApp._make_task_wrapper for W3C context propagation"
+            )
+            original_method = getattr(WorkflowApp, "_make_task_wrapper", None)
+            logger.debug(f"🔍 Original _make_task_wrapper method: {original_method}")
+
+            WorkflowApp._make_task_wrapper = make_context_aware_task_wrapper
+            logger.debug(
+                "✅ Applied W3C context propagation enhancement for Dapr Workflow tasks"
+            )
+
+        except ImportError as e:
+            logger.warning(
+                f"Could not apply W3C context propagation fix (ImportError): {e}"
+            )
+        except Exception as e:
+            logger.error(
+                f"Error applying W3C context propagation fix: {e}", exc_info=True
+            )
+
+    def _apply_agent_wrappers(self) -> None:
+        """
+        Apply observability wrappers for agent execution methods.
+
+        Instruments core agent methods to create AGENT spans (top-level) and CHAIN spans
+        (processing steps) for comprehensive agent execution tracing in Phoenix UI.
+        """
+        # Main agent run wrapper (AGENT span - top level)
+        wrap_function_wrapper(
+            module="dapr_agents.agents.agent.agent",
+            name="Agent.run",
+            wrapper=AgentRunWrapper(self._tracer),
+        )
+
+        # Process iterations wrapper (CHAIN span - processing steps)
+        wrap_function_wrapper(
+            module="dapr_agents.agents.agent.agent",
+            name="Agent.process_iterations",
+            wrapper=ProcessIterationsWrapper(self._tracer),
+        )
+
+        # Tool execution batch wrapper (TOOL span - batch execution)
+        wrap_function_wrapper(
+            module="dapr_agents.agents.agent.agent",
+            name="Agent.execute_tools",
+            wrapper=ExecuteToolsWrapper(self._tracer),
+        )
+
+        # Individual tool execution wrapper (TOOL span - actual tool execution)
+        wrap_function_wrapper(
+            module="dapr_agents.tool.executor",
+            name="AgentToolExecutor.run_tool",
+            wrapper=RunToolWrapper(self._tracer),
+        )
+
+    def _apply_workflow_wrappers(self) -> None:
+        """
+        Apply observability wrappers for workflow orchestration methods.
+
+        Instruments DurableAgent workflow execution to create comprehensive AGENT spans
+        that capture the complete workflow lifecycle from start to completion, enabling
+        detailed monitoring of workflow orchestration in Phoenix UI.
+        """
+        # Workflow execution wrapper for DurableAgent (AGENT span with monitoring)
+        # This captures the complete workflow lifecycle from start to completion
+        wrap_function_wrapper(
+            module="dapr_agents.workflow.base",
+            name="WorkflowApp.run_and_monitor_workflow_async",
```

Comment:
so another PR enabled the sync start with run() on durable agent so should we specify async here?